features/yard-doctest.feature

```diff
@@ -494,6 +494,32 @@ Feature: yard doctest
     When I run `bundle exec yard doctest`
     Then the output should contain "2 runs, 2 assertions, 0 failures, 0 errors, 0 skips"
 
+  Scenario: supports test-name hooks for multiple examples on the same code object
+    Given a file named "doctest_helper.rb" with:
+      """
+      require 'app/app'
+
+      YARD::Doctest.before('#flag') do
+        @flag = true
+      end
+
+      YARD::Doctest.before('#flag#test_0001_Second example for flag') do
```

Comment:
What do you think if we use a different notation, e.g. `#flag@Second example for flag`? I feel like `#test_0001_` in is implementation detail provided by Minitest.

I'm good to merge this anyways if you don't like the proposed changes.